### ISSUE TITLE
Plasma 6 Digital Clock widget is missing KDE-PIM calendar integration

### ISSUE THREAD

**Issue by jansol** [NixOS/nixpkgs#344025]
### Describe the bug

The Digital Clock widget in the plasma status bar shows a calendar when clicked, but the option to show calendars from the KDE PIM suite is missing.

### Steps To Reproduce
```
services.desktopManager.plasma6.enable = true;
programs.kde-pim = {
  enable = true;
  kmail = true;
  kontact = true;
  merkuro = true;
};
```
### Expected behavior
If `programs.kde-pim.enable = true;`, the KDE PIM event source should be listed in available plugins for the calendar of the Digital Clock widget. Or at the very least if Merkuro is also enabled via the corresponding module option.

### Screenshots
![image](https://github.com/user-attachments/assets/e44d0ade-bc48-4337-ad5c-32440336834f)

### Notify maintainers

@K900

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
fetching path input 'path:/nix/store/hiasfhl8f5yy88hcfbr3s8s4bm63wsjw-source'
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.0, NixOS, 24.11 (Vicuna), 24.11.20240919.c04d565`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.0
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/user/.config/nix/nix.conf:/nix/store/1c1580lym6lyjyrc8hidmdrp21zzxwp8-plasma-workspace-6.1.5/etc/xdg/nix/nix.conf:/nix/store/rfr8ffp1nkli4nlhy2a8pxcqn02gbly8-kglobalacceld-6.1.5/etc/xdg/nix/nix.conf:/nix/store/w33p3hrwxdj129sjbzbbqlc7x9f75xsh-baloo-6.6.0/etc/xdg/nix/nix.conf:/home/user/.config/kdedefaults/nix/nix.conf:/nix/store/1c1580lym6lyjyrc8hidmdrp21zzxwp8-plasma-workspace-6.1.5/etc/xdg/nix/nix.conf:/nix/store/rfr8ffp1nkli4nlhy2a8pxcqn02gbly8-kglobalacceld-6.1.5/etc/xdg/nix/nix.conf:/nix/store/w33p3hrwxdj129sjbzbbqlc7x9f75xsh-baloo-6.6.0/etc/xdg/nix/nix.conf:/nix/store/1c1580lym6lyjyrc8hidmdrp21zzxwp8-plasma-workspace-6.1.5/etc/xdg/nix/nix.conf:/nix/store/rfr8ffp1nkli4nlhy2a8pxcqn02gbly8-kglobalacceld-6.1.5/etc/xdg/nix/nix.conf:/nix/store/w33p3hrwxdj129sjbzbbqlc7x9f75xsh-baloo-6.6.0/etc/xdg/nix/nix.conf:/home/user/.config/kdedefaults/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/user/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/user/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/user/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/15q4w2423rnnxxqfbl0mrsjgbpikziik-lix-2.91.0/share`
 - nixpkgs: `/nix/store/hiasfhl8f5yy88hcfbr3s8s4bm63wsjw-source`
```
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by K900** [comment]
It's part of `kdepim-addons`, which is, confusingly, a separate package, and pulls in most of Plasma, which is why it's not enabled by default.

**Comment by jansol** [comment]
Ouch, that is inconvenient...

**Comment by K900** [comment]
We could add an option for `programs.kde-pim.addons` or something, I guess.

**Comment by jansol** [comment]
How much does it add on top of whatever is pulled in by the plasma6 module?

**Comment by K900** [comment]
Not a lot, the problem is that you may want KDEPIM on top of not Plasma.

**Comment by jansol** [comment]
Right, but how much value does `kdepim-addons` have on top of Not Plasma?

**Comment by K900** [comment]
A considerable amount, it's a grab bag of random nonsense that doesn't fit into any other package, so it has plugins for a bunch of other things too.

**Comment by jansol** [comment]
Hmm yea, looks like a bit of a mess... For a lack of easier options, maybe it would make indeed sense to have a `programs.kde-pim.addons` and have it default to true iff the plasma6 module is enabled? Then it would work as expected with Plasma and not clutter Not Plasma unnecessarily.

**Comment by K900** [comment]
If we do that, we should just set `kde-pim.addons.enable` in the Plasma module, as we already do for `kde-pim.enable`. It's better than cursed contextual defaults.

**Comment by jansol** [comment]
Oh, I'd missed that part somehow. Yeah that would make sense then. Although I assume that would then always unconditionally pull in all 3 applications from the `kde-pim` module?

**Comment by K900** [comment]
It doesn't have to, technically? We could just have them in the store and not expose them in the user environment. ~~Or we could chop up the addons package~~

**Comment by jansol** [comment]
I guess splitting up the package is the actual sensible choice. I'll have a look at how bad that could possibly be when I find the time. Could be nice to just bundle the extensions directly with their respective applications where that doesn't introduce new dependencies.

**Comment by monkiki** [comment]
Any current workaroud?

**Comment by K900** [comment]
Install `kdepim-addons`.

**Comment by monkiki** [comment]
Already installed, but I don't see the option in Digital Clock nor see any event in Digital Clock calendar.

**Comment by K900** [comment]
How did you install it?

**Comment by monkiki** [comment]
I tried from command line, but it seems it was already installed. 

Using Kubuntu 24.10

**Comment by K900** [comment]
Uhh, excuse me? Why are you reporting issues from Kubuntu on a NixOS bug tracker?

**Comment by monkiki** [comment]
Sorry, you are true: I didn't realize

**Comment by notjiatan** [comment]
wtf did I just read

**Comment by jansol** [comment]
Same version here, nixos-unstable with akonadi-25.04.3 and akonadictl 6.4.3

**Comment by K900** [comment]
Can someone see if this reproduces on 25.08 and 6.4.4?

**Comment by Nick1296** [comment]
I just updated the latest 25.11, and I can confirm this happens with akonadictl 6.5.0 and 25.08.0, should I get akonadictl 6.4.4?

**Comment by K900** [comment]
No, Plasma 6.4.4 is what you want.

**Comment by Nick1296** [comment]
Ok then yes, I have plasma 6.4.4  on nixos-unstable  (updated my flake just a few hours ago) and there are the same issues I reported

**Comment by K900** [comment]
OK, can you post the contents of your `~/.config/plasma-org.kde.plasma.desktop-appletsrc`?

**Comment by K900** [comment]
Yeah I don't know what `enabledCalendarPlugins=/usr/lib/qt/plugins/plasmacalendarplugins/astronomicalevents.so,/usr/lib/qt/plugins/plasmacalendarplugins/holidaysevents.so,astronomicalevents,holidaysevents,pimevents` is about, but you want to remove those paths from the config.

**Comment by Nick1296** [comment]
Removed the paths, now I only have `enabledCalendarPlugins=holidaysevents,astronomicalevents,pimevents
`  but still no events are shown in the widget.

**Comment by wiessall** [comment]
@Spatchy I have the same setup, syncing Nextcloud calendar to KOrganizer. I had:
```
kdePackages.akonadi
kdePackages.akonadi-import-wizard
kdePackages.kdepim-addons
kdePackages.kdepim-runtime
kdePackages.libkdepim
```
and enabling Calendar Events in the Digital Clock settings displayed my calendars correctly.
 After some updates this didn't work for me anymore. Now (plasmashell 6.4.4 and akonadictl 6.5.0) I also had to add
```kdePackages.akonadi-calendar``` to my config and my events are displayed again in the widget.

**Comment by Nick1296** [comment]
@Spatchy @wiessall Thanks for checking, I was mostly relying on the `programs.kde-pim` option in nixpkgs and I can still confirm that my calendar widget show no calendar events, trying with a local calendar (nixpkgs unstable, akonadictl v6.5.1 and plasmashell 6.4.5)

**Comment by jansol** [comment]
Installing `kdePackages.akonadi-calendar` in addition to `kdePackages.kdepim-addons` does indeed fix it for me. Thanks for the hint @wiessall!

**Comment by eclairevoyant** [comment]
> Elaborate?

Can't add external accounts to `merkuro-{calendar,contact}`. I planned to use Cal/CardDAV but actually _none_ of the integration options are available. Many of the menu options are also broken (do nothing when clicked, or throw a spurious error).

Or if you're asking why I think it should be scrapped, IMO we shouldn't pretend to support things in nixpkgs that we don't support by virtue of simply not testing these scenarios (e.g. merkuro without plasma, apparently).

<img width="402" alt="Image" src="https://github.com/user-attachments/assets/c594d65e-8bc6-4e30-952e-1d70e6f69f00" />

<img width="393" alt="Image" src="https://github.com/user-attachments/assets/fd4ad3a1-a131-462d-9f02-385e95bfbdee" />

> [@eclairevoyant](https://github.com/eclairevoyant) it was working for me not long ago and [these](https://gitlab.com/Nick1296/nixos-conf/-/blob/master/nixos/modules/KDE.nix) are my modules for KDE, take a look especially to the modules that are enabled if kde-pim.enable is true. The module you are looking for should be `akonadi-calendar` but I am not 100% sure

My config was posted above, I have all the packages discussed earlier in this thread. I don't plan to use plasma in any case.

**Comment by digtail** [comment]
I found this issue because merkuro-calendar did not show any calendar source when trying to add a new calendar. after some debugging in the matrix chat the problem has been traced to missing .desktop files. adding this config solved that problem for me.
```
    environment.pathsToLink = [
      "/share/akonadi"
    ];
```

**Comment by eclairevoyant** [comment]
That does seem to work for caldav, but some settings are still unavailable in the application.
